### PR TITLE
syslog/channel: add constant attribute if SYSLOG_IOCTL is not enabled

### PR DESCRIPTION
**Pull request by anchao**
## Summary

syslog/channel: add constant attribute if SYSLOG_IOCTL is not enabled

move all private channel define from data to rodata

Signed-off-by: chao an <anchao@lixiang.com>

## Impact

N/A

## Testing

ci-check

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



**No, this PR does not fully meet the NuttX requirements.**

**Missing Information:**

* **Summary:**
    *  **Why** is this change necessary? What problem does it solve or what improvement does it provide? 
    * **What** functional part of the code is being changed (be specific)?
    * **How** does the change work? Provide a clear explanation of the technical implementation.
* **Impact:**
    * While "N/A" might be accurate for some fields, it's best to explicitly confirm  each point:
       * **Impact on user:**  Will users notice any changes in functionality or behavior? 
       * **Impact on build:** Will any build steps, dependencies, or configurations be affected?
       *  **Impact on documentation:** Even if no documentation updates are required, state this explicitly. 
* **Testing:**
    * **Insufficient Detail:** "ci-check" is not descriptive enough.  Specify:
        *  **Build Hosts:** List the operating systems, CPUs, and compiler versions used for testing.
        *  **Targets:**  Provide details about the target architectures and boards you tested on.
    * **No Logs:**  The "Testing logs" sections are empty. Include relevant logs demonstrating the issue before the change and the improvement after the change.

**Recommendations:**

1. **Expand the Summary:** Provide a more detailed explanation of the change's purpose, implementation, and benefits.
2. **Address all Impact Points:** Explicitly confirm or describe the impact (or lack thereof) for each category.
3. **Provide Specific Testing Details:**  List the build hosts and target environments used for testing. Include relevant log snippets to illustrate the problem and the solution.

File: drivers/syslog/Kconfig [hunk 1]
```diff
@@ -375,4 +377,10 @@ config SYSLOG_IOCTL
 		commands are SYSLOGIOC_SETFILTER/SYSLOGIOC_GETCHANNELS, which can be
 		used to set the enable status of different channels
 
+config SYSLOG_REGISTER
+	bool "Register syslog channel support"
```
Comment (xiaoxiang781216) [inline drivers/syslog/Kconfig:381]: should we remove prompt string and change `depend on` to `select`

Comment (anchao) [reply drivers/syslog/Kconfig:382]: Done

File: drivers/syslog/syslog.h [hunk 2]
```diff
@@ -48,7 +48,11 @@ extern "C"
  */
 
 struct syslog_channel_s; /* Forward reference */
```
Comment (xiaoxiang781216) [inline drivers/syslog/syslog.h:50]: remove

Comment (anchao) [reply drivers/syslog/syslog.h]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**